superduper/misc/auto_schema.py

```diff
@@ -137,6 +133,26 @@ def infer_schema(
     return Schema(identifier=identifier, fields=schema_data)  # type: ignore
 
 
+class VectorTypeFactory(DataTypeFactory):
+    """A factory for Vector datatypes # noqa."""
+
+    @staticmethod
+    def check(data: t.Any) -> bool:
+        """Check if the data is able to be encoded by the JSON serializer.
+
+        :param data: The data object
+        """
+        return isinstance(data, np.ndarray) and len(data.shape) == 1
+
+    @staticmethod
+    def create(data: t.Any) -> BaseDataType | FieldType:
+        """Create a JSON datatype.
+
+        :param data: The data object
+        """
+        return Vector(shape=(len(data),), dtype=str(data.dtype))
```

Comment:
This doesn’t seem to support a list of floats. So for the vector data type, it must be numpy, right? I think it make sense.